tests/test_route.py

```diff
@@ -112,14 +98,12 @@ def test_RouteAddRemoveIpv4Route(self, dvs, testlog):
         time.sleep(1)
 
         # check application database
-        tbl = swsscommon.Table(self.pdb, "ROUTE_TABLE")
-        route_entries = tbl.getKeys()
+        route_entries = self.pdb.get_keys("ROUTE_TABLE")
         assert "2.2.2.0/24" in route_entries
 
         # check ASIC route database
-        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY")
         route_found = False
-        for key in tbl.getKeys():
+        for key in self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY"):
             route = json.loads(key)
             if route["dest"] == "2.2.2.0/24":
                 route_found = True
```

Comment:
I think this is probably OK since you are waiting for the key in ROUTE_TABLE above, but for some extra insurance you may consider doing something like this: https://github.com/Azure/sonic-swss/blob/3052fb5f25fea0ba2f60ad17c294e6da8b9eb70a/tests/test_sub_port_intf.py#L104

Reply:
Updated the check for ASIC_DB into another wait_until function.